src/protocol/response_writer.rs

```diff
@@ -212,6 +212,14 @@ impl<'a, C: Connection + 'a> ResponseWriter<'a, C> {
         Ok(())
     }
 
+    /// Marks data of 'size' bytes as unavailable.
+    pub fn write_x(&mut self, size: usize) -> Result<(), Error<C::Error>> {
```

Comment:
I'd avoid introducing this method without also switching other places in the codebase to use `write_x`. Instead, for this PR, I'd just inline this logic into the call-site, and use `write_str("xx")`

Reply:
I'll take care of this in a separate "refactoring" commit.

BTW, I could only find one other place:
```
$ rgrep -I --include '*.rs' 'write\S*([^x]*x\S*)'
src/protocol/response_writer.rs:                    self.write(b ^ 0x20)?
src/protocol/response_writer.rs:            self.write(b'x')?;
src/stub/core_impl/base.rs:                        res.write_str(";QEnvironmentHexEncoded+")?;
src/stub/core_impl/base.rs:                    res.write_str(";qXfer:exec-file:read+")?;
src/stub/core_impl/base.rs:                    res.write_str(";qXfer:auxv:read+")?;
src/stub/core_impl/section_offsets.rs:                        res.write_str("Text=")?;
src/stub/core_impl/section_offsets.rs:                        res.write_num(text)?;
src/stub/core_impl/section_offsets.rs:                        res.write_str("TextSeg=")?;
src/stub/core_impl/section_offsets.rs:                        res.write_num(text_seg)?;
gdbstub_arch/src/x86/reg/core32.rs:            write_bytes!(&xmm_reg.to_le_bytes());
gdbstub_arch/src/x86/reg/core32.rs:        write_bytes!(&self.mxcsr.to_le_bytes());
gdbstub_arch/src/x86/reg/core64.rs:            write_bytes!(&xmm_reg.to_le_bytes());
gdbstub_arch/src/x86/reg/core64.rs:        write_bytes!(&self.mxcsr.to_le_bytes());
```
did you have other call-sites in mind?

Reply:
Ahh, yeah, would you look at that! I thought it would crop up in more places, but I guess not...

If there were more call sites, I'd be more inclined to introduce this new helper, but given that there are only 2 in the entire codebase, I'd still suggest that we avoid this extra indirection, and inline those calls.

Reply:
`write_x()` is no more!